### PR TITLE
Add pinebookpro build scripts

### PR DESCRIPTION
**Pull request by davidmhewitt**
Here are the build scripts (and associated files) I've been working on in my fork over the past few weeks to build images for the Pinebook Pro.

If merged, this *should* start uploading a daily 6.0 image for the PBP to the same place as where we upload our daily isos.

I'm happy to write up some copy about the limitations and disclaimers surrounding this if needed.

### CODE REVIEW

**Comment by cassidyjames** [comment]
@davidmhewitt I think the biggest thing would be installation instructions. Maybe on the wiki? @btkostner also how will this work with the builds site? Will we be able to differentiate between images somehow?

**Comment by btkostner** [comment]
Ummm, the builds site shows everything in daily with 6.0 in the path. So this will probably cause issues

**Comment by davidmhewitt** [comment]
The filenames will currently come out as something like `elementaryos-6.0-daily-pinebookpro-20200805.img.xz`. So I guess people could possibly download the wrong image if they somehow miss or don't understand the `pinebookpro` bit. Though there'd be no harm in flashing a pinebook pro image to a USB stick and trying to boot it on an x86 machine. It just... wouldn't work.

The alternative is uploading them to a different folder or a subfolder and then we're back to the question of how do we download them.

**Comment by danrabbit** [comment]
On the builds site we should probably show builds from today as a separate list. I imagine there will be other use cases for branches of daily builds as well, like when we had a separate branch for the new installer

**Comment by btkostner** [comment]
We could add a subnav to the builds site for the different "branches" of daily builds. Obviously defaulting to the normal builds.

File: build-pinebookpro.sh [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+apt-get install -y curl python3 python3-distutils
+
+curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+python3 get-pip.py
+pip install boto3
+
+python3 upload.py "$KEY" "$SECRET" "$ENDPOINT" "$BUCKET" "$IMGPATH" "$IMGNAME" || exit 1
+
+CHECKSUMPATH="${basedir}"/${imagename}.md5.txt
+CHECKSUMNAME=${channel}/$(basename "$CHECKSUMPATH")
```
Comment (cassidyjames) [inline build-pinebookpro.sh:372]:
```suggestion
CHECKSUMNAME=${channel}-pinebookpro/$(basename "$CHECKSUMPATH")
```

File: build-pinebookpro.sh [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+sha256sum ${imagename}.img.xz > ${imagename}.sha256.txt
+
+cd "${rootdir}"
+
+KEY="$1"
+SECRET="$2"
+ENDPOINT="$3"
+BUCKET="$4"
+IMGPATH="${basedir}"/${imagename}.img.xz
+IMGNAME=${channel}/$(basename "$IMGPATH")
```
Comment (cassidyjames) [inline build-pinebookpro.sh:361]:
```suggestion
IMGNAME=${channel}-pinebookpro/$(basename "$IMGPATH")
```

File: build-pinebookpro.sh [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+
+python3 upload.py "$KEY" "$SECRET" "$ENDPOINT" "$BUCKET" "$IMGPATH" "$IMGNAME" || exit 1
+
+CHECKSUMPATH="${basedir}"/${imagename}.md5.txt
+CHECKSUMNAME=${channel}/$(basename "$CHECKSUMPATH")
+
+python3 upload.py "$KEY" "$SECRET" "$ENDPOINT" "$BUCKET" "$CHECKSUMPATH" "$CHECKSUMNAME" || exit 1
+
+CHECKSUMPATH="${basedir}"/${imagename}.sha256.txt
+CHECKSUMNAME=${channel}/$(basename "$CHECKSUMPATH")
```
Comment (cassidyjames) [inline build-pinebookpro.sh:377]:
```suggestion
CHECKSUMNAME=${channel}-pinebookpro/$(basename "$CHECKSUMPATH")
```

**Pull request merged**